File: generic.cpp
#include<iostream>
using namespace std;

template <class T>

T Addition(T No1, T No2)
{
   T Ans = 0;
   Ans = No1 + No2;
   return Ans;
}

int main()
{
  int x = 10, y = 11, ret = 0;
  ret = Addition(x,y);
  cout<<"int Addition is :"<<ret<<"\n";

  float a = 10.4, b = 11.8, fret = 0;
  fret = Addition(a,b);
  cout<<"float Addition is :"<<fret<<"\n";

  double d1 = 14.7, d2 = 19.4, dret = 0;
  dret = Addition(d1, d2);
  cout<<"double Addition is :"<<dret<<"\n";

  return 0;
}

File: generic.java
import java.lang.*;

class Arithematic <T>
{
   public T Addition (T No1, T No2)
   {
     T Ans;

     if(No1 instanceof Integer && No2 instanceof Integer)
     {
        return (T)(Integer)((Integer)No1 + (Integer)No2);
     }
     else if(No1 instanceof Float && No2 instanceof Float)
     {
        return (T)(Float)((Float)No1 + (Float)No2);
     }
     else if(No1 instanceof Double && No2 instanceof Double)
     {
        return (T)(Double)((Double)No1 + (Double)No2);
     }
     else
     {
        return null;
     }
   }
}

class Generic
{
   public static void main(String arg[])
   {
      Arithematic <Integer> obj = new Arithematic<Integer>();
      Integer ret = 0;
      ret = obj.Addition(10,11);
      System.out.println("Addition is :"+ret);
   }

}